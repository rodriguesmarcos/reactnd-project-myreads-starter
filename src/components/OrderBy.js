import React from 'react';
import PropTypes from 'prop-types';

const OrderBy = ({onOrderBy, orderBy}) => (
  <div>
    Order by
    <select
      name="orderBy"
      defaultValue="orderBy"
      onChange={(e) => onOrderBy(e.target.value)}
    >
      <option value="title">Title</option>
      <option value="pageCount">Page Count</option>
      <option value="publishedDate">Published Date</option>
    </select>
  </div>
);

OrderBy.propTypes = {
  orderBy: PropTypes.string.isRequired,
  onOrderBy: PropTypes.func.isRequired,
}

export default OrderBy;
