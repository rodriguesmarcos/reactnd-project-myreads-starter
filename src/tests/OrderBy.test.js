import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import OrderBy from '../components/OrderBy.js';
import books from './fixtures/books';
import shelves from './fixtures/shelves';

describe('<OrderBy />', () => {

  const minProps = {
    orderBy: 'title',
    onOrderBy: () => {},
  }

  test('renders without exploding', () => {
    const wrapper = shallow(<OrderBy {...minProps}/>);
    expect(wrapper.length).toBe(1);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('should call onOrderBy', () => {
    const onOrderBySpy = jest.fn();
    const value = 'numPages';
    const wrapper = shallow(<OrderBy {...minProps} onOrderBy={onOrderBySpy}/>);
    wrapper.find('select').simulate('change', {
      target: { value }
    });
    expect(onOrderBySpy).toHaveBeenCalledWith(value);
  });

});
