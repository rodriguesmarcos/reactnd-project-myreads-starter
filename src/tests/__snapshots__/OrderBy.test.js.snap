exports[`<OrderBy /> renders without exploding 1`] = `
<div>
  Order by
  <select
    defaultValue="orderBy"
    name="orderBy"
    onChange={[Function]}>
    <option
      value="title">
      Title
    </option>
    <option
      value="pageCount">
      Page Count
    </option>
    <option
      value="publishedDate">
      Published Date
    </option>
  </select>
</div>
`;
